service/pkg/util/redact.go

```diff
@@ -0,0 +1,65 @@
+package util
+
+import (
+	"reflect"
+	"strings"
+)
+
+func RedactSensitiveData(i interface{}, sensitiveFields []string) interface{} {
+	v := reflect.ValueOf(i)
+	redacted := redact(v, sensitiveFields)
+	return redacted.Interface()
+}
+
+func redact(v reflect.Value, sensitiveFields []string) reflect.Value {
+	//nolint:exhaustive // default case covers other type
+	switch v.Kind() {
+	case reflect.Ptr:
+		if v.IsNil() {
+			return v
+		}
+		redacted := reflect.New(v.Elem().Type())
+		redacted.Elem().Set(redact(v.Elem(), sensitiveFields))
+		return redacted
+	case reflect.Struct:
+		redacted := reflect.New(v.Type()).Elem()
+		for i := 0; i < v.NumField(); i++ {
+			field := v.Field(i)
+			fieldType := v.Type().Field(i)
+			if fieldType.Type.Kind() == reflect.String && contains(sensitiveFields, fieldType.Name) {
+				redacted.Field(i).SetString("REDACTED")
```

Comment:
```suggestion
				redacted.Field(i).SetString("***")
```

This mirrors github action redactions: 

<img width="702" alt="image" src="https://github.com/opentdf/platform/assets/38867245/7d589b26-52cf-4308-8b7d-5c5267b18e2f">